### PR TITLE
[Go] add localvec to correspond to TypeScript dev-local-vectorstore

### PR DESCRIPTION
**Pull request by ianlancetaylor**
Add a genkit.DebugLog functions so that plugins can log debug info.

### CODE REVIEW

File: go/plugins/localvec/localvec.go [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+	"path/filepath"
+	"slices"
+
+	"github.com/google/genkit/go/ai"
+	"github.com/google/genkit/go/genkit"
+)
+
+// New returns a new local vector database. This will register a new
+// retriever with genkit, and also return it.
+// This retriever may only be used by a single goroutine at a time.
```
Comment (jba) [inline go/plugins/localvec/localvec.go:39]: I assume this is based on the equivalent js? Can you add a comment with a reference to the relevant file?

File: go/plugins/localvec/localvec.go [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+	return dot / (l1 * l2)
+}
+
+// docID returns the ID to use for a Document.
+// This is intended to be the same as the genkit Typescript computation.
+func docID(doc *ai.Document) (string, error) {
+	b, err := json.Marshal(doc)
+	if err != nil {
+		return "", fmt.Errorf("pinecone: error marshaling document: %v", err)
```
Comment (jba) [inline go/plugins/localvec/localvec.go:244]: s/pinecone/localvec/

**Review by jba — APPROVED** [review]

File: go/plugins/localvec/localvec.go (hunk 1 above)
Comment (ianlancetaylor) [reply go/plugins/localvec/localvec.go:39]: Done.

File: go/plugins/localvec/localvec.go (hunk 2 above)
Comment (ianlancetaylor) [reply go/plugins/localvec/localvec.go:244]: Whoops, fixed.

**Pull request merged**